toolkit/components/uniffi-bindgen-gecko-js/fixtures/tests/test_rondpoint.js

```diff
@@ -3,25 +3,232 @@
+   Assert.deepEqual(dico, copyDico);
+
+   Assert.equal(await copieEnumeration(Enumeration.DEUX), Enumeration.DEUX);
+   Assert.deepEqual(await copieEnumerations([Enumeration.UN, Enumeration.DEUX]), 
+   [Enumeration.UN, Enumeration.DEUX]);
    const obj = {
-      "zero": zero
+      "0": new EnumerationAvecDonnees.Zero(),
+      "1": new EnumerationAvecDonnees.Un(1),
+      "2": new EnumerationAvecDonnees.Deux(2, "deux")
    };
 
-   const ret = await copieCarte(obj);
-   Assert.equal(enumeration, un);
-   Assert.ok(enumeration === Enumeration.UN);
-   Assert.deepEqual(obj, ret);
-   Assert.deepEqual(enumerations, [un, deux]);
-   Assert.ok(ret.zero instanceof EnumerationAvecDonnees.Zero)
-   Assert.ok(ret.zero instanceof EnumerationAvecDonnees)
-})
+   Assert.deepEqual(await copieCarte(obj), obj);
+
+   const zero = new EnumerationAvecDonnees.Zero();
+   const one = new EnumerationAvecDonnees.Un(1);
+   const two = new EnumerationAvecDonnees.Deux(2);
+   Assert.notEqual(zero, one);
+   Assert.notEqual(one, two);
+
+   Assert.deepEqual(zero, new EnumerationAvecDonnees.Zero());
+   Assert.deepEqual(one, new EnumerationAvecDonnees.Un(1));
+   Assert.notDeepEqual(one, new EnumerationAvecDonnees.Un(4));
+   
+   Assert.ok(await switcheroo(false));
+   // Test the roundtrip across the FFI.
+   // This shows that the values we send come back in exactly the same state as we sent them.
+   // i.e. it shows that lowering from JS and lifting into rust is symmetrical with 
+   //      lowering from rust and lifting into JS.
+
+   const rt = await Retourneur.init();
+
+   const affirmAllerRetour = async (arr, fn, equalFn) => {
+      for (const member of arr) {
+         if (equalFn) {
+            equalFn(await fn(member), member)
+         } else {
+            Assert.equal(await fn(member), member);
+         }
+      }
+   }
+
+   // Booleans
+   await affirmAllerRetour([true, false], rt.identiqueBoolean.bind(rt));
+
+   // Bytes
+   await affirmAllerRetour([-128, 127], rt.identiqueI8.bind(rt));
+   await affirmAllerRetour([0, 0xFF], rt.identiqueU8.bind(rt));
+
+   // Shorts
+   await affirmAllerRetour([-32768, 32767], rt.identiqueI16.bind(rt));
+   await affirmAllerRetour([0, 0xFFFF], rt.identiqueU16.bind(rt));
+
+   // Ints
+   await affirmAllerRetour([0, 1, -1, -2147483648, 2147483647], rt.identiqueI32.bind(rt));
+   await affirmAllerRetour([0, 0xFFFFFFFF], rt.identiqueU32.bind(rt));
+
+
+   // Longs
+   // NOTE: we cannot represent greater than `Number.MAX_SAFE_INTEGER`
+   await affirmAllerRetour([0, 1, -1, Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER], rt.identiqueI64.bind(rt));
+   await affirmAllerRetour([0, Number.MAX_SAFE_INTEGER], rt.identiqueU64.bind(rt));
+
+   // Floats
+   const equalFloats = (a, b) => Assert.ok(Math.abs(a - b) <= Number.EPSILON);
+   await affirmAllerRetour([0.0, 0.5, 0.25, 1.5], rt.identiqueFloat.bind(rt), equalFloats);
+   //   Some float value's precision gets messed up, an example is 3.22, 100.223, etc
```

Comment:
something to keep in mind here, the precision gets slightly off as we roundtripped for numbers like `3.22`, (enough that the `EPSILON` check failed)